resource-managers/nomad/src/main/scala/org/apache/spark/scheduler/cluster/nomad/SparkNomadJobController.scala

```diff
@@ -75,6 +77,31 @@ private[spark] class SparkNomadJobController(jobManipulator: NomadJobManipulator
     }
   }
 
+  def removeExecutors(executorIds: Seq[String]): Future[Boolean] = {
+    try {
+      executorIds
+        .map(f => f.replaceFirst("-\\d+$", ""))
```

Comment:
nomad-spark executor IDs are built as `${allocId}-${epoch}`, so we remove the epoch here to isolate the alloc ID